src/SFA.DAS.Admin.Roatp.Web/Controllers/AddProvider/SelectProviderController.cs

```
@@ -52,14 +56,27 @@ public async Task<IActionResult> Index(AddProviderSubmitModel submitModel, Cance
             return View(viewModel);
         }
 
-        ApiResponse<GetUkrlpResponse> ukrlpResponse = await _outerApiClient.GetUkrlp(int.Parse(submitModel.Ukprn!)!, cancellationToken);
+        ApiResponse<GetUkrlpResponse> ukrlpApiResponse = await _outerApiClient.GetUkrlp(int.Parse(submitModel.Ukprn!)!, cancellationToken);
 
-        if (ukrlpResponse.StatusCode == System.Net.HttpStatusCode.OK)
+        if (ukrlpApiResponse.StatusCode == System.Net.HttpStatusCode.NotFound)
         {
-            return RedirectToRoute(RouteNames.AddProvider);
+            return RedirectToRoute(RouteNames.ProviderNotFoundInUkrlp);
         }
 
-        return RedirectToRoute(RouteNames.ProviderNotFoundInUkrlp);
+        GetUkrlpResponse ukrlpResponse = ukrlpApiResponse.Content!;
```

We need to throw for any other failed status code after handling not found and before assuming success response.